lib/cyclonedx/base.rb

```diff
@@ -1,31 +1,21 @@
 module Cyclonedx
   class Base
-    DEFAULT_COMPONENT_TYPE = "application".freeze
     DEFAULT_DEP_COMPONENT_TYPE = "library".freeze
 
+    attr_accessor :config
+
     def initialize(scan_report, config = {})
       @scan_report = scan_report
       @config = config
     end
 
-    def build_metadata
-      {
-        "timestamp": "",
-        "component": {
-          "bom-ref": "",
-          "type": DEFAULT_COMPONENT_TYPE,
-          "group": "",
-          "name": "",
-          "version": "",
-          "purl": ""
-        }
-      }
-    end
-
     # Returns the 'components' object for a supported/unsupported scanner's report
     def build_components_object
-      components = []
       info = @scan_report.to_h.fetch(:info)
+      return [] unless info[:dependencies]
```

Comment:
If there are no dependencies, would `info[:dependencies]` be `nil` or something like `[]`?

Reply:
Yep `info[:dependencies]` becomes nil which causes errors if the code execution proceeds to ` info[:dependencies].each` leading to an exception